pkg/v4/resource/certconfig/desired.go

```diff
@@ -384,7 +384,7 @@ func newNodeOperatorCertConfig(clusterConfig cluster.Config, cert certs.Cert, na
 				AllowBareDomains:    false,
 				ClusterComponent:    certName,
 				ClusterID:           clusterConfig.ClusterID,
-				CommonName:          clusterConfig.Domain.NodeOperator,
+				CommonName:          clusterConfig.Domain.API,
```

Comment:
Lets please have a TODO comment and a linked issue that we have to get the permissions straight once the guest clusters are set up with the appropriate cluster role bindings.